src/Adapter/Cart/CartPresenter.php

```diff
@@ -45,17 +45,29 @@ protected function presentProduct(array $rawProduct)
             $this->imageRetriever,
             $this->link,
             $this->pricePresenter,
-            new ProductColorsRetriever,
+            new ProductColorsRetriever(),
             $this->translator
         );
 
-        $settings = new ProductPresentationSettings;
+        $settings = new ProductPresentationSettings();
 
         $settings->catalog_mode = Configuration::get('PS_CATALOG_MODE');
         $settings->include_taxes = $this->includeTaxes();
-        $settings->allow_add_variant_to_cart_from_listing =  (int)Configuration::get('PS_ATTRIBUTE_CATEGORY_DISPLAY');
+        $settings->allow_add_variant_to_cart_from_listing = (int) Configuration::get('PS_ATTRIBUTE_CATEGORY_DISPLAY');
         $settings->stock_management_enabled = Configuration::get('PS_STOCK_MANAGEMENT');
 
+        if (is_string($rawProduct['attributes'])) {
```

Comment:
I don't like it, but I don't want to take the risk to update `Cart` class for now.